packages/radix-ui-themes/src/components/data-list.props.ts

```diff
@@ -0,0 +1,105 @@
+import { trimProp } from '../helpers';
+import type { PropDef } from '../helpers';
+
+const alignValues = ['start', 'center', 'end', 'baseline'] as const;
+const gapValues = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'] as const;
+const orientationValues = ['horizontal', 'vertical'] as const;
+const sizes = ['1', '2', '3'] as const;
+const widthValues = ['1', '2', '3'] as const;
+
+const dataListPropDefs = {
+  gap: {
+    type: 'enum | string',
+    className: 'rt-r-gap',
+    customProperties: ['--gap'],
+    values: gapValues,
+    default: '4',
+    responsive: true,
+  },
+  gapX: {
+    type: 'enum | string',
+    className: 'rt-r-cg',
+    customProperties: ['--column-gap'],
+    values: gapValues,
+    default: undefined,
+    responsive: true,
+  },
+  gapY: {
+    type: 'enum | string',
+    className: 'rt-r-rg',
+    customProperties: ['--row-gap'],
+    values: gapValues,
+    default: undefined,
+    responsive: true,
```

Comment:
Decided to not abstract this out yet for flexibility in case we want to tweak the defaults easily.

Reply:
I'd go ahead with abstracting it, safe to say we'll want all gap properties to work the same on all components